### PR TITLE
Update persistence module and add test module

### PR DESCRIPTION
**Pull request by gary-murray-seat**
- rename the repository module to data module
- align the data module package structure with that in the core (all inside infrastructure)
- updated the poms a bit to add all in the management sections
- create a test module and added to IT test (in java )
- add the springboot CampaignCoreApplication class to the core

### CODE REVIEW

File: recall2-campaigns-core/src/main/kotlin/com/seatcode/module/campaign/core/infrastructure/persistence/CampaignPersistenceRepository.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.seatcode.module.campaign.core.infrastructure.persistence
+
+import com.seatcode.module.campaign.core.application.campaign.CreateCampaignCommand
+import com.seatcode.module.campaign.core.domain.model.campaign.CampaignId
+
+interface CampaignPersistenceRepository {
+
+    fun create(createCampaignCommand: CreateCampaignCommand): CampaignId
```
Comment (monro93) [inline recall2-campaigns-core/src/main/kotlin/com/seatcode/module/campaign/core/infrastructure/persistence/CampaignPersistenceRepository.kt:8]: this one is weird for me, I think that the command should not be sent to the repo, instead the interface should be in domain and using the aggregate roots when possible

File: recall2-campaigns-data/src/main/kotlin/com/seatcode/module/campaign/infrastructure/persistence/jpa/AbstractJpaPersistableEntity.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * was updated by someone else before you commit the transaction.
+ *
+ * @param <ID>
+</ID> */
+@MappedSuperclass
+abstract class AbstractJpaPersistableEntity<ID :Serializable> {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    open lateinit var id: ID
```
Comment (monro93) [inline recall2-campaigns-data/src/main/kotlin/com/seatcode/module/campaign/infrastructure/persistence/jpa/AbstractJpaPersistableEntity.kt:19]: I think that we should use uuids, but we should not autogenerate it by db in any case, the id is something important in the domain and should come from there

Comment (gary-murray-seat) [reply recall2-campaigns-data/src/main/kotlin/com/seatcode/module/campaign/infrastructure/persistence/jpa/AbstractJpaPersistableEntity.kt]: done

File: recall2-campaigns-core/src/main/kotlin/com/seatcode/module/campaign/core/infrastructure/persistence/CampaignPersistenceRepository.kt (hunk 1 above)
Comment (gary-murray-seat) [reply recall2-campaigns-core/src/main/kotlin/com/seatcode/module/campaign/core/infrastructure/persistence/CampaignPersistenceRepository.kt]: updated thanks

**Review by monro93 — APPROVED** [review]

**Pull request merged**